packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,326 @@
-/// rendered automatically when this is updated and rendered.
-/// They are translated by this component's (x,y). They do not need to fit
-/// within this component's (width, height).
-abstract class PositionComponent extends BaseComponent {
-  /// The position of this component on the screen (relative to the anchor).
-  Vector2 get position => _position;
-  set position(Vector2 position) => _position.setFrom(position);
-  final Vector2 _position;
-
-  /// X position of this component on the screen (relative to the anchor).
-  double get x => _position.x;
-  set x(double x) => _position.x = x;
-
-  /// Y position of this component on the screen (relative to the anchor).
-  double get y => _position.y;
-  set y(double y) => _position.y = y;
-
-  /// The size that this component is rendered with before [scale] is applied.
-  /// This is not necessarily the source size of the asset.
-  Vector2 get size => _size;
-  set size(Vector2 size) => _size.setFrom(size);
-  final Vector2 _size;
-
-  /// Width (size) that this component is rendered with.
-  double get width => scaledSize.x;
-  set width(double width) => size.x = width / scale.x;
-
-  /// Height (size) that this component is rendered with.
-  double get height => scaledSize.y;
-  set height(double height) => size.y = height / scale.y;
-
-  /// The scale factor of this component
-  Vector2 get scale => _scale;
-  set scale(Vector2 scale) => _scale.setFrom(scale);
-  final Vector2 _scale;
-
-  /// Cache to store the calculated scaled size
-  final ValueCache<Vector2> _scaledSizeCache = ValueCache();
-
-  /// The size that this component is rendered with after [scale] is applied.
-  Vector2 get scaledSize {
-    if (!_scaledSizeCache.isCacheValid([scale, size])) {
-      _scaledSizeCache.updateCache(
-        size.clone()..multiply(scale),
-        [scale.clone(), size.clone()],
-      );
+/// The main properties of this class is the [_transform] (which combines
+/// the [position], [angle] of rotation, and [scale]), the [size], and
+/// the [anchor]. Thus, the [PositionComponent] can be imagined as an
+/// abstract picture whose size originally is [size]; within that picture
+/// a point at location [anchor] is selected, that point is designated as
+/// a logical "center" of the picture. Then, the transform is applied to
+/// the picture: its anchor is moved to the point [position] on the screen,
+/// then the picture is rotated by [angle] around the anchor point, and
+/// scaled by [scale] also around the anchor point.
+///
+/// The [size] property of the [PositionComponent] is used primarily for
+/// tap and collision detection. Thus, the [size] should be set equal to
+/// the approximate bounding rectangle of the rendered picture. If you
+/// do not specify the size of a PositionComponent, then it will be
+/// equal to zero and the component won't be able to respond to taps.
+///
+class PositionComponent extends BaseComponent {
```

Comment:
Good that you removed `abstract`, I did that in my PR too, but this will probably be merged first.

Reply:
Yeah, I was following your lead here :) Someone would have to update [this picture](https://flame-engine.org/docs/#/components) though

Reply:
Indeed, I'll update that with the other PR since the game classes will be put in there too.